src/udx.c

```diff
@@ -453,24 +452,20 @@ clear_outgoing_packets (udx_stream_t *stream) {
     }
   }
 
-  // also clear pending unordered packets, and the destroy packet if waiting
-  udx_fifo_t *u = &(stream->unordered);
-
-  while (u->len > 0) {
-    udx_packet_t *pkt = udx__fifo_shift(u);
+  while (stream->unordered_queue.len > 0) {
+    udx_packet_t *pkt = udx__queue_data(udx__queue_shift(&stream->unordered_queue), udx_packet_t, queue);
     if (pkt == NULL) continue;
 
-    if (pkt->type == UDX_PACKET_TYPE_STREAM_SEND) {
-      udx_stream_send_t *req = pkt->ctx;
+    assert(pkt->type == UDX_PACKET_TYPE_STREAM_SEND);
 
-      if (req->on_send != NULL) {
-        req->on_send(req, UV_ECANCELED);
-      }
-    }
+    udx_stream_send_t *req = pkt->ctx;
+    assert((void *) req == (void *) pkt);
 
-    if (pkt->type & UDX_PACKET_FREE_ON_SEND) {
```

Comment:
this is free is safe to remove. the only packet type put in the unordered queue are packet UDX_PACKET_TYPE_STREAM_SEND, which is not free-on-send.